### ISSUE TITLE
R Session Aborted when running the RNN example

### ISSUE THREAD

**Issue by Freestyleyang** [joshuamarie/kindling#12]
Hi Joshua,

Thank you very much for developing the amazing package. I have been looking for an elegant way to build a LSTM model in R for a long time, and I truly appreciate your efforts. 

I am running the RNN example, however, when I run the code, an error occurs "R Session Aborted". I have tried to reduce the hidden neurons, but the error still occurs. I wonder if you could help me with this? Thanks. 

The example I run is below for your reference. I am using R 4.4.1 and RStudio 2025.09.1. 

```
library(mlbench)
library(tidymodels)
library(kindling)

data(Ionosphere, package = "mlbench")
ionosphere_data = Ionosphere[, -2]

rnn_kindling(mode = "classification", 
             hidden_neurons = c(16, 8), 
             activations = act_funs(relu, elu), 
             epochs = 50, 
             rnn_type = "gru") %>% 
  fit(Class ~ ., data = ionosphere_data) %>% 
  augment(new_data = ionosphere_data) %>% 
  metrics(truth = Class, estimate = .pred_class)
```

<img width="769" height="505" alt="Image" src="https://github.com/user-attachments/assets/c5a9cd3f-5ae1-4e82-9c43-8c6877141154" />

In addition, the parameter tuning example uses:
```
mlp_tune_spec = mlp_kindling(mode = "classification", 
                             hidden_neurons = tune(), 
                             activations = tune(), 
                             output_activation = tune())

iris_folds = vfold_cv(iris, v = 3)

nn_wf = workflow() %>% 
  add_recipe(recipe(Species ~ ., data = iris)) %>% 
  add_model(mlp_tune_spec)

nn_grid_depth = grid_depth(hidden_neurons(c(4L, 16L)), 
                           activations(c("relu", "elu")), 
                           output_activation(c("sigmoid", "linear")), 
                           n_hlayer = 2, 
                           size = 10, 
                           type = "latin_hypercube")

nn_tunes = tune_grid(nn_wf, iris_folds, grid = nn_grid_depth)
```

However, `output_activation` only accepts c("relu", "elu", "selu", "softplus", "softmax", "log_softmax", "logsigmoid", "hardtanh", "hardsigmoid", "silu"). Thus, "linear" is not a valid option, and an error occurs below. Could you please confirm? Thank you. 

<img width="787" height="70" alt="Image" src="https://github.com/user-attachments/assets/a400fbb2-fae2-40b4-8524-ef2e640427eb" />

**Comment by joshuamarie** [comment]
"R Session Aborted" problem happens within your R session, not on the package. Simply because you ran too many things in 1 session. 

Try run this example:

```r
box::use(
    kindling[rnn_kindling, act_funs],
    parsnip[fit, augment],
    yardstick[metrics]
)

data(Ionosphere, package = "mlbench")
ionosphere_data = Ionosphere[, -2]

rnn_kindling(mode = "classification",
             hidden_neurons = c(16, 8),
             activations = act_funs(relu, elu),
             epochs = 50,
             rnn_type = "gru") |>
    fit(Class ~ ., data = ionosphere_data) |>
    augment(new_data = ionosphere_data) |>
    metrics(truth = Class, estimate = .pred_class)
``` 

Store it in a script, then execute the script through the following command:

```shell
rscript ./name_of_your_script.R
``` 

If you are in Windows and `rscript` doesn't found, place your executable (binary) of your R into system environment variables. 

This is what happens when I run the following code in bash:

```bash
$ rscript ./try.R
# A tibble: 2 × 3
  .metric  .estimator .estimate
  <chr>    <chr>          <dbl>
1 accuracy binary         0.989
2 kap      binary         0.975
```

**Comment by joshuamarie** [comment]
As for the hyperparameter tuning, that caused by one of the parameters: `optimizer`. We got the same occurrence, and somehow it worked (there's no failure in `R CMD check`, and perhaps we could ignore it). Worry not, it is part of our to-do list.

**Comment by Freestyleyang** [comment]
Hi Joshua,

Thank you very much for your reply. Running the script in R terminal has fixed the issue! 

I wonder if it is possible to run the script using RStudio in the future? It would be much more convenient to do all the data analysis (e.g., feature engineering) and RNN training in one place. Thanks.

**Comment by joshuamarie** [comment]
> I wonder if it is possible to run the script using RStudio in the future? It would be much more convenient to do all the data analysis (e.g., feature engineering) and RNN training in one place. Thanks.

All of (if not, most of) IDEs out there, including RStudio, already has this. In your case, you ran too many things, and maybe, minimize it a bit.
